lib/dai-plugin-mcd/src/SavingsService.js

```diff
@@ -0,0 +1,80 @@
+import { PublicService } from '@makerdao/services-core';
+import { ServiceRoles } from './constants';
+import { MDAI } from './index';
+import BigNumber from 'bignumber.js'
+import { RAY, WAD } from './constants'
+
+export default class SavingsService extends PublicService {
+  constructor(name = ServiceRoles.SAVINGS) {
+    super(name, ['smartContract', 'proxy', 'accounts', ServiceRoles.SYSTEM_DATA]);
+  }
+
+  async join(amountInDai) {
+    await this.get('proxy').ensureProxy();
+
+    const chi = await this.chi();
+    const amount = amountInDai.toBigNumber().div(chi);
```

Comment:
Afaict the proxy actions contract [divides by `chi` for us](https://github.com/makerdao/dss-proxy-actions/blob/9f9e275a55fa93f675698f2fd1a105b7f2c74e2e/src/DssProxyActions.sol#L497) before calling `pot.join`, so we should be able to pass `MDAI(amountInDai).toFixed('wei')` straight into `_proxyActions.dsrJoin` :)

(ditto for `exit`)

Reply:
Dang I keep missing stuff like that. Good catch!